### PR TITLE
DASH 400 error fix for Android

### PR DESCRIPTION
**Pull request by juan-cortes**
After a bit of 🕵🏼 detective work it seems that our explorers will not consider the payload from a POST as valid if we don't pass the correct content type header. This was used as a default in the _pre-okhttp3_ implementation for all posts, which caused other issues. This approach will attempt to check if the payload is a valid JSON string and if it is, will add the appropriate header.

Seems to fix the DASH send 400 error that was detected by @Arnaud97234 yesterday.

I'm creating another pr against master but perhaps we want to cherry-pick the commit 
694fe5a if 065f9d2 is already merged somewhere else and I'm missing context cc @gre

### CODE REVIEW

**Comment by gre** [comment]
@juan-cortes unfortunately you need to target hotfixes/4.x because master is not stable (libcore 4.2) can you resolve conflicts?

**Review by gre — CHANGES_REQUESTED** [review]
conflictts to target the previous work.

**Comment by juan-cortes** [comment]
I resolved the conflicts, it should be fine now 🤔

**Comment by juan-cortes** [comment]
No, wait, this still has all the 5.x stuff, let me try again 😆

**Comment by juan-cortes** [comment]
Should be fine now 🤞🏼 @gre

**Review by gre — APPROVED** [review]
